solver/src/metrics.rs

```diff
@@ -278,6 +299,17 @@ impl SolverMetrics for Metrics {
         self.matched_but_unsettled_orders.inc_by(count as u64);
     }
 
+    fn report_order_surplus(
+        &self,
+        winning_solver: &str,
+        best_alternative: &str,
+        surplus_diff: f64,
+    ) {
+        self.order_surplus_report
+            .with_label_values(&[winning_solver, best_alternative])
```

Comment:
One downside is that this will create a loooooot of permutations (`n * (n - 1)` where `n` is the number of solvers, so 42 in prod).

Maybe, we may consider not using labels at all for this metric and rely on logs when this does happen.

On the other hand it would be cool to measure things like how often solver X beat out Y but didn't get picked. However, maybe Prometheus metrics is not the right way to measure this kind of thing.

Don't have strong opinions either way, will differ to other's judgement on this.

Reply:
Yes. It would be much simpler if we just report the diffs although then we could only use this for alerting and have to rely on the logs. I agree having this many new tiles in the dashboard would be silly and not so easy to look at, but from a data perspective it would be useful and we wouldn't need to actually use the labels in the grouping for our grafana alert.

Reply:
Sounds good. Lets get rid of the labels for now then.

Reply:
I read

> Lets get rid of the labels for now then.

as saying there wouldn't be labels. That is what I agreed with. I am against abusing Prometheus for use cases it wasn't made for - like high dimensional data. I think it makes sense as a single dimensional metric for alerts like you said in previous post.

Reply:
Right, although I had suggested that the grafana metrics would only ever be single dimensional since we would ignore the labels, but I also read Nicks comment wrong and thought he was agreeing with keeping them and not using them.

Will remove the labels for now, but I have a feeling they will come back in some way or another.